scripts/update_package_versions.py

```diff
@@ -292,7 +298,9 @@ def _get_ambiguous_public_ids(
             operator.itemgetter(0),
             filter(
                 lambda x: x[1] > 1,
-                Counter(id_.public_id for id_ in all_package_ids_to_update).items(),
+                Counter(
+                    _extract_prefix(id_.public_id) for id_ in all_package_ids_to_update
```

Comment:
now two public ids are ambiguous if they have the same prefix. Otherwise, it might lead to wrong updates (e.g. public ids updated twice if the conflicting ids have consecutive semantic versions (e.g. `fetchai/ml_data_provider:0.10.0` and `fetchai/ml_data_provider:0.11.0`).

Reply:
alternative: detect this corner case, and process the public ids from the greatest version to the smallest.

Reply:
No; we cannot do that because processing order MUST be determined by the package type; not the package name!